rust_src/src/windows.rs

```diff
@@ -1074,6 +1074,34 @@ pub fn window_new_total(window: LispWindowValidOrSelected) -> LispObject {
     win.new_total
 }
 
+/// Set new total size of WINDOW to SIZE.
+/// WINDOW must be a valid window and defaults to the selected one.
+/// Return SIZE.
+///
+/// Optional argument ADD non-nil means add SIZE to the new total size of
+/// WINDOW and return the sum.
+///
+/// The new total size of WINDOW, if valid, will be shortly installed as
+/// WINDOW's total height (see `window-total-height') or total width (see
+/// `window-total-width').
+///
+/// Note: This function does not operate on any child windows of WINDOW.
+#[lisp_fn(min = "2")]
+pub fn set_window_new_total(
+    window: LispWindowValidOrSelected,
+    size: EmacsInt,
+    add: Option<LispObject>,
```

Comment:
It looks like `add` can simply be a `bool`

Reply:
It is the best that `add` is `LispObject`, I think.

The rust's `bool` cannot be put here because `add` is an optional arg. 
There's no implement for `From(Lispobject) for Some(bool)`, so `Some(bool)` cannot be here.
`Option<LispObject>` need checking `None` and checking `add.is_nil()`.
If this is `LispObject`, `add.is_nil()` is `nil` either there is no optional args or args is nil.

Reply:
I am ok with it staying `LispObject`. But you are perhaps confused about how the optional arguments are handled.

Qnil is passed as the argument if one is not provided, otherwise some other LispObject value is used. So the `From<bool>` in lisp.rs will work just find. Qnil is False and anything else is True.

For many of our functions we use `Option<T>`. Qnil becomes `None` and anything else becomes Some(T). The LispObject having a value of Qnil because the user did not provide or they sent `nil` on purpose both end up being the same `None`. Typically, the `as_foo()` method of `LispObject` is called to create a `Foo`. For instance if the function takes an optional LispWindow then `LispObject.as_window()` is called by the `lisp_fn` macro via the `Option<LispWindow>` From implementation.

Reply:
Perhaps this is a place where our documentation could use an improvement? If so, would you like to take a try at updating it?